File: backend/app/models/enums.py
"""Shared enums for the LOKAL domain."""
from enum import Enum


class UserMode(str, Enum):
    """Runtime mode toggle — a user can flip between these any time."""
    CONSUMER = "consumer"
    PROVIDER = "provider"


class ServiceCategory(str, Enum):
    """Top-level provider categories. A user can hold N of these
    simultaneously — the whole point of LOKAL."""
    RIDE_DELIVERY = "ride_delivery"
    TECHIE = "techie"
    SUPPORT_PARTNER = "support_partner"
    NON_TECH = "non_tech"


class VerificationStatus(str, Enum):
    PENDING = "pending"
    SUBMITTED = "submitted"
    VERIFIED = "verified"
    REJECTED = "rejected"


class RequestStatus(str, Enum):
    DRAFT = "draft"
    OPEN = "open"
    MATCHED = "matched"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

File: app/lib/core/models/enums.dart
/// Domain enums — mirrored from backend/app/models/enums.py.
/// Keep these in lock-step with the server.

enum UserMode {
  consumer,
  provider;

  String get value => name;
  static UserMode fromString(String? v) =>
      UserMode.values.firstWhere((m) => m.name == v, orElse: () => UserMode.consumer);
}

enum ServiceCategory {
  rideDelivery('ride_delivery', 'Ride & Delivery'),
  techie('techie', 'Techie'),
  supportPartner('support_partner', 'Support Partner'),
  nonTech('non_tech', 'Non-Tech Services');

  const ServiceCategory(this.value, this.label);
  final String value;
  final String label;

  static ServiceCategory? fromValue(String? v) {
    for (final c in ServiceCategory.values) {
      if (c.value == v) return c;
    }
    return null;
  }
}

enum VerificationStatus {
  pending,
  submitted,
  verified,
  rejected;

  String get value => name;
  static VerificationStatus fromString(String? v) => VerificationStatus.values
      .firstWhere((s) => s.name == v, orElse: () => VerificationStatus.pending);
}

enum RequestStatus {
  draft,
  open,
  matched,
  inProgress('in_progress'),
  completed,
  cancelled;

  const RequestStatus([String? override]) : _override = override;
  final String? _override;
  String get value => _override ?? name;
}
